### PR TITLE
Add documentation for TempConfig.

### PR DESCRIPTION
**Pull request by karliss**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/developers-docs/first-time.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/developers-docs.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

**Test plan (required)**

Screenshot from 2020-05-30 12-48-04

**Closing issues**

### CODE REVIEW

File: src/common/TempConfig.h [hunk 1]
```diff
@@ -5,6 +5,23 @@
...
+ *
+ * \code
+ * {
+ *     TempConfig tempConfig;
+ *     tempConfig.set("asm.arch", "x86").set("asm.comments", false);
+ *     return Core()->cmdRaw("pd");
+ *     // config automatically restored at the end of scope
+ * }
+ * \endcode
+ */
```
Comment (ITAYC0HEN) [inline src/common/TempConfig.h:24]: Maybe better to demonstrate the before-scope-after workflow?

Comment (karliss) [reply src/common/TempConfig.h:24]: I am not sure what you meant by that.

Comment (ITAYC0HEN) [reply src/common/TempConfig.h:24]:
```
 
// asm.comments is set to true
// Disassemble 3 lines with comments
qInfo() << Core()->cmdRaw("pd 3");

{
    TempConfig tempConfig;
    tempConfig.set("asm.arch", "x86").set("asm.comments", false);   

    // Disassemble 3 lines without comments
    return Core()->cmdRaw("pd 3");

    // config automatically restored at the end of scope
}

// Disassemble 3 lines with comments
qInfo() << Core()->cmdRaw("pd 3");
```

Comment (karliss) [reply src/common/TempConfig.h:24]: That seems excessive. Not affecting past shouldn't have to be explained. I also don't like having in an example more extra code than code directly using class it is explaining.

**Pull request merged**